lib/db.js

```diff
@@ -76,14 +75,19 @@ module.exports = (
     }
 
     this.pool = new Pool(options.url, pooleeOptions)
+    this.redis = require('./redis')(config.redis, log)
   }
 
   DB.connect = function (options) {
     return P.resolve(new DB(options))
   }
 
   DB.prototype.close = function () {
-    return P.resolve(this.pool.close())
+    const promises = [this.pool.close()]
+    if (this.redis) {
+      promises.push(this.redis.close())
```

Comment:
Ah, nice one! This should also help us update mocha fwiw. The latest version sits there hanging until all async listeners have resolved....